File: Editor/src/main/java/org/turbomedia/turboedit/editor/renderer/RenderServerConnectionManager.java
package org.turbomedia.turboedit.editor.renderer;

import java.util.HashMap;

public class RenderServerConnectionManager {
    private static final HashMap<String, RenderServerConnection> connections = new HashMap<>();

    public static RenderServerConnection EstablishConnection(RenderServerEntry entry) {
        if (connections.containsKey(entry.id())) {
            return connections.get(entry.id());
        }

        var connection = new RenderServerConnection(entry, new RendererClient(entry));

        connections.put(entry.id(), connection);

        return connection;
    }

    public static RenderServerConnection getConnection(String id) {
        return connections.get(id);
    }
}

File: Editor/src/main/kotlin/turboedit/editor/renderer/RenderServerConnectionManager.kt
package turboedit.editor.renderer

object RenderServerConnectionManager {
    private val connections = HashMap<String?, RenderServerConnection?>()

    @JvmStatic
    fun establishConnection(entry: RenderServerEntry): RenderServerConnection? {
        if (connections.containsKey(entry.id())) {
            return connections.get(entry.id())
        }

        val connection = RenderServerConnection(entry, RendererClient(entry))

        connections.put(entry.id(), connection)

        return connection
    }

    fun getConnection(id: String?): RenderServerConnection? {
        return connections.get(id)
    }
}
